src/backend/executor/nodeDML.c

```diff
@@ -172,16 +178,28 @@ ExecDML(DMLState *node)
 			Assert(!isnull);
 		}
 
-		/* Correct tuple count by ignoring deletes when splitting tuples. */
-		ExecDelete(tupleid,
-				   segid,
-				   NULL, /* GPDB_91_MERGE_FIXME: oldTuple? */
-				   node->cleanedUpSlot,
-				   NULL /* DestReceiver */,
-				   node->ps.state,
-				   !isUpdate, /* GPDB_91_MERGE_FIXME: where to get canSetTag? */
-				   PLANGEN_OPTIMIZER /* Plan origin */,
-				   isUpdate);
+		if (DML_DELETE == action)
+			/* Correct tuple count by ignoring deletes when splitting tuples. */
```

Comment:
I think, it is better to use braces here, or even do something
```c
if (DML_DELETE != action)
{
	ExecUpdate(...);
	return slot;
}
```
and then leave delete block unchanged